docs/06_how-to-guides/40_multi-index/how-to-define-a-primary-index.md

```diff
@@ -2,76 +2,124 @@
+  +    name test_primary;
+  +    // additional data stored in table row, e.g. an uint64_t type data
+  +    uint64_t datum;
+    };
+  ```
+
+### 3. Define The Primary Index
+
+Add definition of the primary index for the multi-index table. The primary index type must be uint64_t, it must be unique and it must be named `primary_key()`, otherwise the compiler (eosio-cpp) will generate an error saying it can not find the field to use as the primary key:
+
+  ```diff
+    // the data structure which defines each row of the table
+    struct [[eosio::table]] test_table {
+      // this property stores a name for each row of the multi-index table
+      name test_primary;
+      // additional data stored in table row
+      uint64_t datum;
+  +    // mandatory definition for primary key getter
+  +    uint64_t primary_key( ) const { return test_primary.value; }
+    };
+  ```
 
 [[info | Secondary indexes information]]
-| Other, secondary, indexes if they will be defined can have duplicates. You can have up to 16 additional indexes and the field types can be uint64_t, uint128_t, uint256_t, double or long double.
-
-5. For ease of use, define a type alias `test_tables` based on the `eosio::multi_index` template type, parametarized with a random name `"testtaba"` and the `test_table` data structure defined above
-```diff
-  // the data structure which defines each row of the table
-  struct [[eosio::table]] test_table {
-    // this field stores a name for each row of the multi index table
-    name test_primary;
-    // additional data stored in table row
-    uint64_t datum;
-    // mandatory definition for primary key getter
-    uint64_t primary_key( ) const { return test_primary.value; }
-  };
-  
-+  typedef eosio::multi_index<"testtaba"_n, test_table> test_tables;
-```
-
-Declare the multi index table as a data member of type `test_tables`, as defined above.
-```diff
-  // the data structure which defines each row of the table
-  struct [[eosio::table]] test_table {
-    // this field stores a name for each row of the multi index table
-    name test_primary;
-    // additional data stored in table row
-    uint64_t datum;
-    // mandatory definition for primary key getter
-    uint64_t primary_key( ) const { return test_primary.value; }
-  };
-  
-  typedef eosio::multi_index<"testtaba"_n, test_table> test_tables;
-+  test_tables testtab;
-```
-
-Now you have instantiated the `testtab` as a multi index table which has a primary index defined for its `test_primary` data member.
+| Other, secondary, indexes if they will be defined can have duplicates. You can have up to 16 additional indexes and the corresponding property types can be uint64_t, uint128_t, uint256_t, double or long double.
+
+### 4. Define A Multi-Index Type Alias
+
+For ease of use, define a type alias `test_table_t` based on the `eosio::multi_index` template type, parametarized with a random name `"testtaba"` and the `test_table` data structure defined above.
```

Comment:
a type alias is a short name for a quite long type definition.
it is easier to use everywhere in the code the alias (which is short) instead of the long type definition.